src/React/Basic/Hooks.purs

```diff
@@ -359,6 +370,13 @@ foreign import memo_ ::
     (ReactComponent props)
     (ReactComponent props)
 
+foreign import memoEq_ ::
+  forall props.
+  EffectFn2
+    (ReactComponent props)
+    (Fn2 props props Boolean)
+    (ReactComponent props)
+
```

Comment:
:information_source:  It might be possible only use a single exports, but since the arity is different this way was easier to implement and maybe is easier to reason about.

One could also think about implementing `memo` in terms of `memo'` but passing passing PureScript's equality function would change the behavior compared to `React.memo`. And passing `undefined`/`null` might work, but is in theory not the same as calling the react function with 1 instead of 2 arguments. Therefore I went with separating both cases.

Reply:
I think this was the right approach 👍🏻